### PR TITLE
add capability to save bounding box in xmin,ymin,xmax,ymax format

### PR DESCRIPTION
**Pull request by Guilhen**
This PR adds the capability to save the bounding box in a traditional format as opposed to in a normalized one.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Enhanced output format options for detection results in YOLOv5.

### 📊 Key Changes
- Added a `use_xyxy` option within the detection script `detect.py`.
- Resulting detection coordinates can now be saved in the `xmin, ymin, xmax, ymax` format.
- Modified the logic in the result-writing part of the code to accommodate the new output format.

### 🎯 Purpose & Impact
- 🎯 **Purpose**: Offers flexibility in the choice of output coordinate formats, catering to different post-processing needs or integration requirements.
- 🚀 **Impact**: Users gaining more control over how detection results are formatted, potentially simplifying integration with other tools or systems that require bounding box coordinates in `xmin, ymin, xmax, ymax` format.

### CODE REVIEW

**Comment by glenn-jocher** [comment]
@Guilhen 👋 Hello! Thanks for asking about **handling inference results**. YOLOv5 🚀 [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) models allow for simple model loading and inference in a pure python environment without using `detect.py`. This includes the option to export boxes in any format, i.e. results.xyxy, results.xywh, and normalized verisons results.xyxyn, results.xywhn:

https://github.com/ultralytics/yolov5/blob/4870064629e76d9d387578c562a292cc680fa05f/models/common.py#L596-L599

### Simple Inference Example

This example loads a pretrained YOLOv5s model from PyTorch Hub as `model` and passes an image for inference. `'yolov5s'` is the YOLOv5 'small' model. For details on all available models please see the [README](https://github.com/ultralytics/yolov5#pretrained-checkpoints). Custom models can also be loaded, including custom trained PyTorch models and their [exported](https://docs.ultralytics.com/yolov5/tutorials/model_export) variants, i.e. ONNX, TensorRT, TensorFlow, OpenVINO YOLOv5 models.

```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')  # or yolov5m, yolov5l, yolov5x, etc.
# model = torch.hub.load('ultralytics/yolov5', 'custom', 'path/to/best.pt')  # custom trained model

# Images
im = 'https://ultralytics.com/images/zidane.jpg'  # or file, Path, URL, PIL, OpenCV, numpy, list

# Inference
results = model(im)

# Results
results.print()  # or .show(), .save(), .crop(), .pandas(), etc.

results.xyxy[0]  # im predictions (tensor)
results.pandas().xyxy[0]  # im predictions (pandas)
#      xmin    ymin    xmax   ymax  confidence  class    name
# 0  749.50   43.50  1148.0  704.5    0.874023      0  person
# 2  114.75  195.75  1095.0  708.0    0.624512      0  person
# 3  986.00  304.00  1028.0  420.0    0.286865     27     tie
```


See [YOLOv5 PyTorch Hub Tutorial](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) for details.

Good luck 🍀 and let us know if you have any other questions!

**Comment by Guilhen** [comment]
Thank you for your response, it makes a lot of sense when the use of PyTorch hub is an option.
It is not always the case for mobile applications and in this case, we have to use the detect.py.

**Comment by glenn-jocher** [comment]
@Guilhen PyTorch Hub is intended for almost all custom solutions, including deployed cloud APIs, and supports TensorRT inference for the fastest speeds on GPU, as well as ONNX and OpenVINO inference for the fastest speeds on CPU. PyTorch Hub is recommended for all custom workflows, and naturally already includes the features submitted in this PR plus additional normalized formats.

detect.py is merely intended as an easy demo for newcomers to run inference on a variety of sources.

**Comment by glenn-jocher** [comment]
@Guilhen I've updated the [PyTorch Hub tutorial](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) with a section explaining how to use exported models here.

Screenshot 2022-05-13 at 12 12 23

**Comment by glenn-jocher** [comment]
@Guilhen thank you for your efforts on this. I'm closing the PR as the updates here are not fully aligned with our roadmap for this part of the code.

I appreciate the time you put in on this though and encourage you to contribute new fixes and features you discover in the future. Thank you for your contributions to YOLOv5 🚀 and Vision AI ⭐

**Comment by Guilhen** [comment]
Hi Glen, thanks for the update, it is very valuable to mention that PyTorch Hub supports more than just the PyTorch inference.

**Comment by glenn-jocher** [comment]
@Guilhen you're welcome! Indeed, PyTorch Hub offers a wide range of capabilities beyond simple PyTorch inference. Feel free to explore and leverage the various functionalities it provides for your specific use cases. Should you have any further questions or need assistance, don't hesitate to ask.
